semgrep-core/src/engine/Match_search_rules.ml

```diff
@@ -260,6 +262,7 @@ let matches_of_patterns ?range_filter config file_and_more patterns =
   in
   match xlang with
   | Xlang.L (lang, _) ->
+      (* TODO can we only force this once? *)
```

Comment:
What is this TODO?

Reply:
I was wondering if we could restructure where we choose to parse the file now